connection.go

```diff
@@ -104,29 +107,50 @@ func (c *Connection) Receive() (dg *Datagram, err error) {
 	// return dg, err
 }
 
-// Queries the given identifier on the RCT device, returning its value as a datagram
-func (c *Connection) Query(id Identifier) (dg *Datagram, err error) {
-	if dg, ok := c.cache.Get(id); ok {
-		return dg, nil
-	}
-	c.builder.Build(&Datagram{Read, id, nil})
-	_, err = c.Send(c.builder)
-	if err != nil {
-		return nil, err
-	}
-
-	dg, err = c.Receive()
-	if err != nil {
-		return nil, err
-	}
-	if dg.Cmd != Response || dg.Id != id {
-		return nil, RecoverableError{fmt.Sprintf("invalid response to read of %08X: %v", id, dg)}
-	}
-	c.cache.Put(dg)
-
-	return dg, nil
+// Queries the given identifier on the RCT device with retry, returning its value as a datagram
+func (c *Connection) Query(id Identifier) (*Datagram, error) {
```

Comment:
Why does this complexity need be part of the library? It alrady signals a recoverable error for retry purposes.

Reply:
From my point of view, the purpose behind incorporating retry logic directly into the library is twofold: to standardize error handling and to simplify application development. This built-in mechanism ensures consistent resilience against transient errors - which occur with RCT devices on a very frequent basis -, reducing the burden on users to implement their own error recovery strategies. Although it adds some complexity, it ultimately enhances the library's robustness and user-friendliness, offering a more reliable and streamlined experience for all.